.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,33 +1,42 @@
-## Description
+<!-- Note: replace the instructions with your text -->
 
-<!-- Just describe what you did. -->
+## Context
 
-<!-- Skip if the title of the PR is self-explanatory -->
+- Describe the objective or issue this PR addresses, as well as the impact of the change.
+- Try to keep the description accessible to newcomers.
+- If you're resolving a specific issue, add "Fixes #issue_number" or "Closes #issue_number".
 
-### Linked issue
+### Solution
 
-<!-- Duplicate the main issue and add additional issues closed by this PR. -->
+- Describe the approach taken to achieve the objective / resolve the issue.
 
-Closes #{issue_number} <!-- Replace with an actual number,  -->
+### Migration Guide (optional)
 
-<!-- Link if e.g. JIRA issue or  from another repository -->
+- If this PR contains a breaking change *(DECIDE: relative to `main` or last release?)*, provide an instruction on how affected parties might need to adapt to the change.
```

Comment:
I don't understand how "relative to latest release" should work.. Can you describe how would this rule work?

Reply:
I'm not sure, but I was thinking about the case where after some release multiple breaking changes happened on the same functionality, overriding one another. The end-user should only know about the sum of the breaking changes. However, breaking changes in Iroha are not just meant for users, but also for SDK devs, DevOps and other parties.

We could compare with the main branch, and then fold overlapping breaking changes manually for users when preparing release notes. I'm unaware of better strategies to handle this, would need to do more research.

Reply:
I think it should be compared against `main`

Reply:
we're looking for breaking changes within the context of the PR. We don't expect contributors to understand what they are breaking compared to the latest release, but they can understand what they are breaking within their PR.

Also, these two categories will overlap more often than not